src/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -228,6 +228,57 @@ public void TestManyOutputLines()
             Assert.Equal(ExpectedLineCount + 1, totalLinesReceived);
         }
 
+        [Fact]
+        public void TestClosingStreamsAsyncDoesNotThrow()
+        {
+            Process p = CreateProcessPortable(RemotelyInvokable.WriteLinesAfterClose);
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+
+            // On netfx, the handler is called once with the EventArgs as null, even if the process writes nothing to the pipe.
+            // That's fine, so we'll ignore it here.
```

Comment:
You could assert PlatformDetection.IsFullFramework so we don't start somehow doing this in .NET Core, if you want  eg 

```c#
p.OutputDataReceived += (s, e) => { if (!(PlatformDetection.IsFullFramework && e == null)) Assert.True(false, "OutputDataReceived called after closing the process"); };
```

Reply:
Better,
```c#
p.OutputDataReceived += (s, e) => { Assert.True(PlatformDetection.IsFullFramework && e == null, "OutputDataReceived called after closing the process"); };
```